test/fields.uts

```diff
@@ -183,6 +183,13 @@ assert p.test2 == 0xc000ff3333
 assert p.test3 == 0xffeeddccbbaa9988776655
 assert p.test4 == 309404098707666285700277845
 
+class TestFuzzNBytesField(Packet):
+    fields_desc = [
+        NBytesField('test1', 0, 128),
+    ]
+
+f = fuzz(TestFuzzNBytesField())
+assert raw(f)[:-1] != raw(f)[:-1]
```

Comment:
I used this test to make sure that it always failed without this PR. With this PR applied it's much less likely to fail but it's still possible I guess. I'll try to figure out how to make it pass all the time. One option would probably be to make sure that the whole range is used without actually unrolling the packets.

Reply:
Did you try using `random.seed()`? It helps stabilize such test, but results sometimes change between Python interpreters.

Reply:
I think I made the test stable by using
```sh
f.test1.max == 2 ** (128 * 8) - 1
```
instead. If fails without this PR because `f.test1.max` is 255 and I think it should be enough to test that the whole range is used. I haven't pushed it yet though.

Reply:
pypy3.9 does not seem happy =/ Otherwise congrats are other tests are passing perfectly.

Reply:
I'll try to take a closer look but as far as I can see the test failure doesn't seem to have anything to do with this PR. The "GMLAN_InitDiagnostics" test also failed in https://github.com/secdev/scapy/pull/3912. I've just re-run the test suite against the master branch and it failed there too: https://github.com/evverx/scapy/actions/runs/4266423827/jobs/7495413171:
```
>>> res = GMLAN_InitDiagnostics(isotpsock, timeout=0.1, retry=1, unittest=True) == False
Sending <GMLAN  service=DisableNormalCommunication |>
<GMLAN  service=NegativeResponse |<GMLAN_NR  requestServiceId=DisableNormalCommunication returnCode=SubFunctionNotSupported deviceControlLimitExceeded=0 |>>
Sending <GMLAN  service=DisableNormalCommunication |>
<GMLAN  service=NegativeResponse |<GMLAN_NR  requestServiceId=DisableNormalCommunication returnCode=SubFunctionNotSupported deviceControlLimitExceeded=0 |>>
>>> thread.join(timeout=5)
>>> assert res
>>> assert ecusimSuccessfullyExecuted == True
Traceback (most recent call last):
  File "<input>", line 2, in <module>
AssertionError
``` 
@polybassa could you take a look? It seems to come from the automotive test suite.

Reply:
Looks like the automotive test suite has nothing to do with the timeouts. The tests pass with https://github.com/secdev/scapy/commit/669506bd42e4141718374ba297974881fe125fb8 reverted (https://github.com/secdev/scapy/pull/3927) so it seems to be the culprit. Apologies!